### PR TITLE
Bug 1499141 - Convert window title to react

### PR DESCRIPTION
**Pull request by camd**

### CODE REVIEW

File: ui/job-view/context/Pushes.jsx [hunk 1]
```diff
@@ -305,8 +305,6 @@ export class PushesClass extends React.Component {
         { pushList: [...pushList], oldestPushTimestamp },
         () => this.setRevisionTips(pushList),
       );
-      this.$rootScope.firstPush = pushList[0];
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/job-view/context/Pushes.jsx]: Good catch!

File: ui/job-view/JobView.jsx [hunk 2]
```diff
@@ -90,8 +90,7 @@ class JobView extends React.Component {
 
     RepositoryModel.getList().then((repos) => {
       const currentRepo = repos.find(repo => repo.name === repoName) || this.state.currentRepo;
-      // To support the title string of the tab when a single revision is showing.
-      this.$rootScope.currentRepo = currentRepo;
+
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/job-view/JobView.jsx]: Having an empty line between ``const`` and the operational parts of a function is a style that Eli encouraged me to follow.  I think I *usually* do, but have noticed a few places where I slipped up and forgot.  It doesn't seem as visually useful to have that empty line when it's a 2-line function, but <shrug>...  :)

File: ui/helpers/revision.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+import unescape from 'lodash/unescape';
+
+import { thTitleSuffixLimit } from './constants';
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/helpers/revision.js]: Good call, yeah.  I just moved the ``const`` into that file.

File: ui/helpers/revision.js [hunk 4]
```diff
@@ -8,3 +12,26 @@ export const parseAuthor = function parseAuthor(author) {
 export const isSHAorCommit = function isSHAorCommit(str) {
   return /^[a-f\d]{12,40}$/.test(str) || str.includes('hg.mozilla.org');
 };
+
+export const getRevisionTitle = function getRevisionTitle(revisions) {
+  let title;
+  for (const revision of revisions) {
+    title = unescape(revision.comments);
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/helpers/revision.js]: Yeah, that gave me pause for a second, too.  But I passed it over thinking it was "safer" somehow.  But on second thought, I think you're right in that we don't need it any longer.  Removing.  :)

[the remark this answers is not in the document]
Comment (edmorley) [reply ui/helpers/revision.js]: Yeah I thought that too at first glance - but safer would be `escape` rather than `unescape` I think?

**Pull request merged**